feat: add accessor protocol support and refactor `stats/base/nanvariancech`

rahulptl165 opened the pull request:
…riancech

Resolves #5674 .

## Description

> What is the purpose of this pull request?
Add accessor protocol support and refactored `@stdlib/stats/base/nanvariancech`.

This pull request:

-   {{TODO: add description describing what this pull request does}}
-  Add new file `lib/accessors.js`.
-  modify `lib/ndaaray.js`.
-  modify `lib/nanvariancech`.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5674 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

rahulptl165 committed:
Add accessor protocol support and refactored @stdlib/stats/base/nanvariancech

aayush0325 commented on lib/node_modules/@stdlib/stats/base/nanvariancech/lib/accessors.js:
```diff
@@ -0,0 +1,126 @@
+
+	if ( N <= 0 ) {
+		return NaN;
+	}
```
```suggestion
```
this is already being checked for in `ndarray.js`

aayush0325 commented on lib/node_modules/@stdlib/stats/base/nanvariancech/lib/accessors.js:
```diff
@@ -0,0 +1,126 @@
+
+	// Find an estimate for the mean...
+	for ( i = 0; i < N; i++ ) {
+		v = xget( xbuf, ix);
```
```suggestion
		v = xget( xbuf, ix );
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/nanvariancech/examples/index.js:
https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/stats/base/dnanvariancepn/examples/index.js

kindly refer to this to see how we write examples. also, use `generic` data type instead of `float64`

aayush0325 commented on lib/node_modules/@stdlib/stats/base/nanvariancech/docs/types/test.ts:
why are you deleting this file...

aayush0325 commented on lib/node_modules/@stdlib/stats/base/nanvariancech/README.md:
````diff
@@ -134,41 +131,35 @@ Note that indexing is relative to the first index. To introduce an offset, use [
 ```javascript
-var floor = require( '@stdlib/math/base/special/floor' );
-
 var x = [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0 ];
````
```suggestion
var x = [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN, NaN ];
```

aayush0325 replied:
we should explicitly show how we're handling the `NaN` element

aayush0325 commented on lib/node_modules/@stdlib/stats/base/nanvariancech/README.md:
```diff
@@ -134,41 +131,35 @@ Note that indexing is relative to the first index. To introduce an offset, use [
-var N = floor( x.length / 2 );
 
-var v = nanvariancech.ndarray( N, 1, x, 2, 1 );
+var v = nanvariancech.ndarray( 4, 1, x, 2, 1 );
```
```suggestion
var v = nanvariancech.ndarray( 5, 1, x, 2, 1 );
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/nanvariancech/README.md:
```diff
@@ -196,18 +188,11 @@ var v = nanvariancech.ndarray( N, 1, x, 2, 1 );
-    x[ i ] = round( (randu()*100.0) - 50.0 );
-}
+var x = normal( 10, 0, 1 );
+x[4] = NaN;
```
same comments as `examples/index.js`

aayush0325 commented on lib/node_modules/@stdlib/stats/base/nanvariancech/benchmark/benchmark.js:
https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/stats/base/dnanvariancepn/benchmark/benchmark.js

kindly use this as a reference

aayush0325 replied:
same comments for your other benchmark file

rahulptl165 committed:
refactor: refactor example.js, benchmark files, readme and accessors.js

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

rahulptl165 commented:
Hi @aayush0325, i have implemented the requested changes. Please let me know if you need any further modification.

aayush0325 commented:
kindly address the CI failure errors before further review, this saves a lot of time for both of us! thank you :)

rahulptl165 committed:
style: fix missing traling slashes

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

rahulptl165 commented:
Hi @aayush0325, I’m sorry for any confusion caused. this branch was removed to address some concerns. A new PR #6167  is now available with the requested changes.

Sorry again !
